pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -135,15 +135,20 @@ def fetch_mirror(remote):
 
 def fetch_remote_url(remote):
     """Fetch a single remote from which can be content synced."""
-    remote_url = remote.url.rstrip("/") + "/"
-    downloader = remote.get_downloader(url=urlpath_sanitize(remote_url, "repodata/repomd.xml"))
-    try:
-        downloader.fetch()
-    except (ClientResponseError, FileNotFoundError):
-        return fetch_mirror(remote)
-    else:
+
+    def normalize_url(url):
+        return url.rstrip("/") + "/"
+
+    remote_url = normalize_url(remote.url)
+    if get_repomd_file(remote, remote_url):
         return remote_url
 
+    remote_url = fetch_mirror(remote)
+    if remote_url:
+        return normalize_url(remote_url)
+
+    raise ValueError(_("An invalid remote URL was provided."))
```

Comment:
Do we want to add remote.url to this message, so it's immediately clear what the invalid-url was?